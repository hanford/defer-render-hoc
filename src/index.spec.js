let mockRaf;
jest.mock("raf", () => {
  return jest.fn(cb => {
    mockRaf = cb;
  });
});

import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import DeferRender from "./";

const Test = ({ name }) => <h1>{name}</h1>;
Test.foo = "bar";
const TestWrappedComponent = DeferRender(Test);

it("should hoist Static Methods from Wrapped Component", () => {
  expect(TestWrappedComponent.foo).toEqual(Test.foo);
});

it("should render Wrapped Component after two animation frames", () => {
  const wrapper = shallow(<TestWrappedComponent name="Abe" />);
  expect(toJSON(wrapper)).toMatchSnapshot();

  mockRaf();
  wrapper.update();
  expect(toJSON(wrapper)).toMatchSnapshot();

  mockRaf();
  wrapper.update();
  expect(toJSON(wrapper)).toMatchSnapshot();
});

