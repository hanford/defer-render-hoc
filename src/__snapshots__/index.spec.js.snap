// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Wrapped Component after two animation frames 1`] = `""`;

exports[`should render Wrapped Component after two animation frames 2`] = `""`;

exports[`should render Wrapped Component after two animation frames 3`] = `
<Test
  name="Abe"
/>
`;
